core/src/main/java/org/apache/iceberg/BaseEntriesTable.java

```diff
@@ -94,6 +105,188 @@ static CloseableIterable<FileScanTask> planFiles(
+      }
+
+      @Override
+      public <T> Boolean ltEq(BoundReference<T> ref, Literal<T> lit) {
+        return compareDateFileContent(ref, lit, compareResult -> compareResult <= 0);
+      }
+
+      @Override
+      public <T> Boolean gt(BoundReference<T> ref, Literal<T> lit) {
+        return compareDateFileContent(ref, lit, compareResult -> compareResult > 0);
+      }
+
+      @Override
+      public <T> Boolean gtEq(BoundReference<T> ref, Literal<T> lit) {
+        return compareDateFileContent(ref, lit, compareResult -> compareResult >= 0);
+      }
+
+      @Override
+      public <T> Boolean eq(BoundReference<T> ref, Literal<T> lit) {
+        return compareDateFileContent(ref, lit, compareResult -> compareResult == 0);
+      }
+
+      @Override
+      public <T> Boolean notEq(BoundReference<T> ref, Literal<T> lit) {
+        return compareDateFileContent(ref, lit, compareResult -> compareResult != 0);
+      }
+
+      @Override
+      public <T> Boolean in(BoundReference<T> ref, Set<T> literalSet) {
+        if (isDateFileContent(ref)) {
+          if (!literalSet.contains(manifestContentId)) {
+            return ROWS_CANNOT_MATCH;
+          }
+        }
+        return ROWS_MIGHT_MATCH;
+      }
+
+      @Override
+      public <T> Boolean notIn(BoundReference<T> ref, Set<T> literalSet) {
+        if (isDateFileContent(ref)) {
+          if (literalSet.contains(manifestContentId)) {
+            return ROWS_CANNOT_MATCH;
+          }
+        }
+        return ROWS_MIGHT_MATCH;
+      }
+
+      @Override
+      public <T> Boolean startsWith(BoundReference<T> ref, Literal<T> lit) {
+        return ROWS_MIGHT_MATCH;
+      }
+
+      @Override
+      public <T> Boolean notStartsWith(BoundReference<T> ref, Literal<T> lit) {
+        return ROWS_MIGHT_MATCH;
+      }
+
+      /**
+       * Comparison of data file content and literal, using integer comparator.
+       *
+       * @param ref bound reference, comparison attempted only if reference is for date_file.content
+       * @param lit literal value to compare with date file content.
+       * @param desiredResult function to apply to int comparator result, returns true if result is
+       *     as expected.
+       * @return false if comparator does not achieve desired result, true otherwise
+       */
+      private <T> Boolean compareDateFileContent(
+          BoundReference<T> ref, Literal<T> lit, Function<Integer, Boolean> desiredResult) {
+        if (isDateFileContent(ref)) {
+          Literal<Integer> intLit = lit.to(Types.IntegerType.get());
+          int literalValueToCompare = toManifestContentId(intLit.value());
+          int cmp = intLit.comparator().compare(manifestContentId, literalValueToCompare);
+          if (!desiredResult.apply(cmp)) {
+            return ROWS_CANNOT_MATCH;
+          }
+        }
+        return ROWS_MIGHT_MATCH;
+      }
+
+      private <T> boolean isDateFileContent(BoundReference<T> ref) {
```

Comment:
wrong spelling 'date'.

Let's call this method 'fileContent'  I think dataFile content was also a confusing part of Iceberg code (probably because the class was first called DataFile and later Content was added to also let it support DeleteFile...).  This also makes much code shorter as this method is used often.